modules/nlu/src/backend/election/natural-election.ts

```diff
@@ -45,20 +48,47 @@ function electIntent(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstandin
   }
 }
 
-function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
-  if (!input.predictions) {
+export function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
+  if (!isNotNil(input.predictions)) {
     return input
   }
 
-  const elected = input.intent!
+  const elected = input.intent
+  if (!isNotNil(elected)) {
+    return input
+  }
 
-  if (!elected) {
+  const electedContext = input.predictions[elected.context]
+  if (!isContextPrediction(electedContext)) {
     return input
   }
 
-  const electedIntent = input.predictions[elected.context].intents.find(i => i.label === elected.name)
-  if (!electedIntent) {
+  const electedContextIntents = electedContext.intents
+  if (!Array.isArray(electedContextIntents)) {
+    sdk.logger.warn(
+      `Warning in ${
+        extractElectedIntentSlot.name
+      } function: 'electedContext.intents' should be an array. ${JSON.stringify(electedContextIntents)}`
+    )
+    return input
+  }
+
+  const maybeInvalidIntentPrediction = electedContextIntents.some(ec => !isIntentPrediction(ec))
+  if (maybeInvalidIntentPrediction) {
```

Comment:
same comment goes for here: Let's either fix the problem at the source or change the typings

Reply:
Not sure I understand?

Fixing the problem at the source will be outside the scope of this PR, especially since there might be none. This is merely type-checking to make the process more robust.